core/storage/suggestion/gae_models.py

```diff
@@ -0,0 +1,262 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for Oppia suggestions."""
+
+from core.platform import models
+import feconf
+
+from google.appengine.ext import ndb
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
+
+# Constants defining types of entities to which suggestions can be created.
+TARGET_TYPE_EXPLORATION = 'exploration'
+TARGET_TYPE_QUESTION = 'question'
+TARGET_TYPE_SKILL = 'skill'
+TARGET_TYPE_TOPIC = 'topic'
+
+TARGET_TYPE_CHOICES = [
+    TARGET_TYPE_EXPLORATION,
+    TARGET_TYPE_QUESTION,
+    TARGET_TYPE_SKILL,
+    TARGET_TYPE_TOPIC
+]
+
+# Constants defining the different possible statuses of a suggestion.
+STATUS_ACCEPTED = 'accepted'
+STATUS_IN_REVIEW = 'review'
+STATUS_INVALID = 'invalid'
+STATUS_REJECTED = 'rejected'
+
+STATUS_CHOICES = [
+    STATUS_ACCEPTED,
+    STATUS_IN_REVIEW,
+    STATUS_INVALID,
+    STATUS_REJECTED
+]
+
+# Constants defining various suggestion types.
+SUGGESTION_EDIT_STATE_CONTENT = 'edit_exploration_state_content'
+
+SUGGESTION_TYPE_CHOICES = [
+    SUGGESTION_EDIT_STATE_CONTENT
+]
+
+# Defines what is the minimum role required to review suggestions
+# of a particular type.
+SUGGESTION_MINIMUM_ROLE_FOR_REVIEW = {
+    SUGGESTION_EDIT_STATE_CONTENT: feconf.ROLE_ID_EXPLORATION_EDITOR
+}
+
+# Constants defining various contribution types.
+SCORE_TYPE_CONTENT = 'content'
+SCORE_TYPE_TRANSLATION = 'translation'
+
+SCORE_TYPE_CHOICES = [
+    SCORE_TYPE_CONTENT,
+    SCORE_TYPE_TRANSLATION
+]
+
+# The delimiter to be used in the suggestion ID.
+SUGGESTION_ID_DELIMITER = '.'
+# The delimiter to be used in score category field.
+SCORE_CATEGORY_DELIMITER = '.'
+
+
+class SuggestionModel(base_models.BaseModel):
+    """Model to store suggestions made by Oppia users.
+
+    The ID of the suggestions are created by joining target_type, target_id and
+    thread_instance_ID using the SUGGESTION_ID_DELIMITER.
+    """
+
+    # The type of suggestion.
+    suggestion_type = ndb.StringProperty(
+        required=True, indexed=True, choices=SUGGESTION_TYPE_CHOICES)
+    # The type of the target entity which the suggestion is linked to.
+    target_type = ndb.StringProperty(
+        required=True, indexed=True, choices=TARGET_TYPE_CHOICES)
+    # The ID of the target entity being suggested to.
+    target_id = ndb.StringProperty(required=True)
+    # The version number of the target entity at the time of creation of the
+    # suggestion.
+    target_version_at_submission = ndb.IntegerProperty(required=True)
+    # Status of the suggestion.
+    status = ndb.StringProperty(
+        required=True, indexed=True, choices=STATUS_CHOICES)
+    # The ID of the author of the suggestion.
+    author_id = ndb.StringProperty(required=True, indexed=True)
+    # The ID of the reviewer assigned to review the suggestion.
+    assigned_reviewer_id = ndb.StringProperty(required=False, indexed=True)
```

Comment:
Drop "required=False" altogether if the field isn't required. That's the default.

Ditto below.

Reply:
Done.